ledger/participant-integration-api/src/test/suite/scala/platform/apiserver/configuration/IndexStreamingCurrentLedgerConfigurationSpec.scala

```diff
@@ -0,0 +1,128 @@
+      when(index.lookupConfiguration())
+        .thenReturn(Future.successful(Some(offset("0001") -> currentConfiguration)))
+
+      IndexStreamingCurrentLedgerConfiguration
+        .owner(index, ledgerConfiguration)
+        .use { currentLedgerConfiguration =>
+          currentLedgerConfiguration.ready.map { _ =>
+            currentLedgerConfiguration.latestConfiguration should be(Some(currentConfiguration))
+            succeed
+          }
+        }
+    }
+
+    "stream the latest configuration from the index" in {
+      val configurations = List(
+        offset("000a") -> Configuration(
+          generation = 3,
+          timeModel = LedgerTimeModel.reasonableDefault.copy(maxSkew = Duration.ofMinutes(1)),
+          maxDeduplicationTime = Duration.ofDays(1),
+        ),
+        offset("0023") -> Configuration(
+          generation = 4,
+          timeModel = LedgerTimeModel.reasonableDefault.copy(maxSkew = Duration.ofMinutes(2)),
+          maxDeduplicationTime = Duration.ofDays(1),
+        ),
+        offset("01ef") -> Configuration(
+          generation = 5,
+          timeModel = LedgerTimeModel.reasonableDefault.copy(maxSkew = Duration.ofMinutes(2)),
+          maxDeduplicationTime = Duration.ofHours(6),
+        ),
+      )
+      val configurationEntries = configurations.zipWithIndex.map {
+        case ((offset, configuration), index) =>
+          offset -> ConfigurationEntry.Accepted(s"submission ID #$index", configuration)
+      }
+      val ledgerConfiguration = LedgerConfiguration(
+        initialConfiguration = Configuration(0, LedgerTimeModel.reasonableDefault, Duration.ZERO),
+        initialConfigurationSubmitDelay = Duration.ZERO,
+        configurationLoadTimeout = Duration.ofSeconds(5),
+      )
+
+      val index = mock[IndexConfigManagementService]
+      when(index.lookupConfiguration()).thenReturn(Future.successful(None))
+      when(index.configurationEntries(None))
+        .thenReturn(Source(configurationEntries).concat(Source.never))
+
+      IndexStreamingCurrentLedgerConfiguration
+        .owner(index, ledgerConfiguration)
+        .use { currentLedgerConfiguration =>
+          currentLedgerConfiguration.ready.map { _ =>
+            eventually {
+              currentLedgerConfiguration.latestConfiguration should be(Some(configurations.last._2))
+            }
+            succeed
+          }
+        }
+    }
+
+    "give up waiting if the configuration takes too long to appear" in {
+      val index = mock[IndexConfigManagementService]
+      when(index.lookupConfiguration()).thenReturn(Future.successful(None))
+      when(index.configurationEntries(None)).thenReturn(Source.never)
+
+      val ledgerConfiguration = LedgerConfiguration(
+        initialConfiguration = Configuration(0, LedgerTimeModel.reasonableDefault, Duration.ZERO),
+        initialConfigurationSubmitDelay = Duration.ZERO,
+        configurationLoadTimeout = Duration.ofMillis(500),
```

Comment:
I think this timeout can be further reduced in this case (1ms or less).

Reply:
I have some upcoming work to use [a fake scheduler](https://doc.akka.io/api/akka/2.6/akka/testkit/ExplicitlyTriggeredScheduler.html) rather than real timeouts in these tests, which makes the actual value of this timeout meaningless (because we advance time programmatically).

Are you OK if we make those changes instead?

Reply:
Absolutely.